doc/connectivity/bluetooth/api/mesh/od_cli.rst

```diff
@@ -3,7 +3,53 @@
 On-Demand Private Proxy Client
 ##############################
 
-On-Demand Private Proxy Client model is used to set and retrieve the On-Demand Private GATT Proxy state, and to adjust message transmission timeout.
+The On-Demand Private Proxy Client model is used to set and retrieve the On-Demand Private GATT Proxy state
+and to adjust message transmission timeout. The State defines how long a node will advertise Mesh Proxy Service with
+Private Network Identity type after it receives Solicitation PDU.
+
+The On-Demand Private GATT Proxy Client model is an optional mesh subsystem that can be enabled through
+the :kconfig:option:`CONFIG_BT_MESH_OD_PRIV_PROXY_CLI` configuration option.
+
+The On-Demand Private Proxy Client communicates with a On-Demand Private Proxy Server
+node that is a recipient of Solicitation PDUs.
+
+Accessing a target's On-Demand Private GATT Proxy state
+*******************************************************
+
+The On-Demand Private Proxy Client can fetch a target's state using the :c:func:`bt_mesh_od_priv_proxy_cli_get` call:
+
+.. code-block:: C
+
+      uint8_t state_rsp;
+
+      const struct bt_mesh_msg_ctx my_ctx = {
+         .addr = 0x0004,
+         .net_idx = 0,
+         .send_ttl = BT_MESH_TTL_DEFAULT,
+      };
+
+      bt_mesh_od_priv_proxy_cli_get(&my_ctx, &state_rsp);
+
+and set a target's state using the :c:func:`bt_mesh_od_priv_proxy_cli_set` call:
+
+.. code-block:: C
+
+      uint8_t new_val = 100;
+      bt_mesh_od_priv_proxy_cli_set(&my_ctx, new_val, &state_rsp);
```

Comment:
I'm not sure if we need such level of details. For such kind of functionality, it should be clear how to use the API. I think snippets are useful for more complex scenarios.